kernel/device.c

```diff
@@ -177,3 +179,45 @@ void device_busy_clear(struct device *busy_dev)
 	ARG_UNUSED(busy_dev);
 #endif
 }
+
+int device_lock_timeout(struct device *dev, k_timeout_t timeout)
+{
+#ifdef CONFIG_DEVICE_CONCURRENT_ACCESS
+	struct device_context *dc =
```

Comment:
i am not convinced that we want to have access control context for all devices. It takes ram and for many devices it is not needed because there is a single user or other methods to maintain control. I'm also pushing for allowing using some devices from interrupt context (i2c/spi) where access control is maintained using queue operation manager (#24781). Imo, it would be better to have pointer to that context in the device structure. null meaning no access control.

Reply:
Yes, I noted that in this RFC message, see "Important notice". 

If there is no resistance in generalizing this concepts, I will improve the RFC so it will be possible to allocate sync/lock semaphores per-instance basis and not to all.